mono/metadata/threads.c

```diff
@@ -4485,24 +4497,42 @@ static void CALLBACK dummy_apc (ULONG_PTR param)
 #endif
 
 /*
- * mono_thread_execute_interruption
+ * mono_thread_execute_interruption_handle
  * 
  * Performs the operation that the requested thread state requires (abort,
  * suspend or stop)
  */
-static MonoException*
-mono_thread_execute_interruption (void)
+static MonoExceptionHandle
+mono_thread_execute_interruption_handle (void)
 {
-	MonoInternalThread *thread = mono_thread_internal_current ();
-	MonoThread *sys_thread = mono_thread_current ();
+	HANDLE_FUNCTION_ENTER ();
 
+	// This function unnecessarily uses the pinned internal thread
+	// via a handle, in order to take advantage of field accessors in handle.h,
+	// against managed pointers within the internal thread.
+	//
+	// Because internal threads are always pinned, a raw pointer
+	// is also used for non-pointer accesses, e.g. integer state, lock, unlock.
+
+	MonoExceptionHandle exc = { 0 };
+	MonoInternalThread* thread = mono_thread_internal_current ();
+	MonoInternalThreadHandle thread_handle = mono_thread_internal_current_handle ();
```

Comment:
I'm not a big fan of both `thread` and `thread_handle` aliasing the same managed object in the same function. Makes it harder to keep track of the state of the object, IMO.

Reply:
This was a dilemma for me.
I wrote it with all handles first.
I was trying to balance change and correctness
and consistency/readability. Since the rule is pretty
clear, raw for ints, handles for pointers, I thought
the result readable. But I can change it.

Reply:
+1 for https://github.com/mono/mono/pull/6639/files#r163305932